### PR TITLE
Delete the User Metadata cache when running clean_user_cache().

### PR DESCRIPTION
**Pull request by dd32**
This is intended as a simple version of #1776

Trac ticket:  https://core.trac.wordpress.org/ticket/54316

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by spacedmonkey — APPROVED** [review]

**Comment by dd32** [comment]
Just noting that I've force-pushed this around a little to run a unit test that fails (for users, passes for posts) first, and then the original commit on top of that showing the unit test passing.

Posts passed originally as `clean_post_cache()` calls the same `wp_cache_delete()`, see https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/post.php#L7373

File: tests/phpunit/tests/meta/updateMetadata.php [hunk 1]
```diff
@@ -27,4 +27,71 @@ public function test_slashed_key_for_existing_metadata() {
 		$found = get_metadata( 'post', 123, 'foo\foo', true );
 		$this->assertSame( 'baz', $found );
 	}
+
+	/**
+	 * @ticket 54316
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/meta/updateMetadata.php:32]:
```suggestion
	 * @ticket 54316
	 *
	 * @group user
	 *
	 * @covers ::clean_user_cache
	 *
	 * @global wpdb $wpdb WordPress database abstraction object.
```

:1234: This also applies to the docblock of the test below. I've used the ordering suggested by @SergeyBiryukov in [WP#56070](https://core.trac.wordpress.org/ticket/56070).

**Review by peterwilsoncc — COMMENTED** [review]
LGTM based on a visual review.

I've added some docblock suggestions inline and confirmed that `wp_cache_delete( $user_id, 'user_meta' );` only follows `clean_user_cache()` in deprecated functions so they can remain unchanged.

File: tests/phpunit/tests/meta/updateMetadata.php [hunk 2]
```diff
@@ -27,4 +27,71 @@ public function test_slashed_key_for_existing_metadata() {
...
+		// Clear the user caches.
+		clean_user_cache( $user_id );
+
+		// Verify metadata cache was cleared.
+		$found = get_metadata( 'user', $user_id, 'key', true );
+		$this->assertSame( 'value2', $found );
+	}
+
+	/**
+	 * @ticket 54316
```
Comment (spacedmonkey) [inline tests/phpunit/tests/meta/updateMetadata.php:71]:
```suggestion
	 * @ticket 54316
	 *
	 * @group user
	 *
	 * @covers ::clean_user_cache
	 *
	 * @global wpdb $wpdb WordPress database abstraction object.
```

**Review by spacedmonkey — APPROVED** [review]

**Comment by spacedmonkey** [comment]
Committed

**Pull request closed without merging**